lib/SPIRV/PreprocessMetadata.cpp

```diff
@@ -256,6 +256,35 @@ void PreprocessMetadataBase::visit(Module *M) {
           .add(getMDOperandAsInt(SchedulerTargetFmaxMhzINTEL, 0))
           .done();
     }
+
+    // !{void (i32 addrspace(1)*)* @kernel, i32 ip_interface, i32 interface}
+    if (MDNode *Interface =
+            Kernel.getMetadata(kSPIR2MD::IntelFPGAIPInterface)) {
+      std::set<std::string> InterfaceStrSet;
+      // Default mode is 'csr' aka !ip_interface !N
+      //                           !N = !{!”csr”}
+      // don't emit any particular SPIR-V for it
+      // Streaming mode metadata be like:
+      // Not 'stall free' mode (to be mapped on '0' literal)
+      // !ip_interface !N
+      // !N = !{!"streaming"}
+      // 'stall free' mode (to be mapped on '1' literal)
+      // !ip_interface !N
+      // !N = !{!"streaming", !"stall_free_return"}
+      int32_t InterfaceMode = -1;
+      for (size_t I = 0; I != Interface->getNumOperands(); ++I)
+        InterfaceStrSet.insert(getMDOperandAsString(Interface, I));
+      if (InterfaceStrSet.find("stall_free_return") != InterfaceStrSet.end())
+        InterfaceMode = 1;
```

Comment:
Should we check here that "streaming" is also present in MD?

Reply:
Hm, actually, lets check it, thanks!